### PR TITLE
Add a build_hash to routeInfo.json files so they can be cached

### PR DESCRIPTION
**Pull request by Rusfighter**
During production, add a build_hash to routeInfo.json so it can be cached by browser for infinite time

https://github.com/nozzle/react-static/issues/717

## Description
More cache control (same caching behaviour as webpack provides with bundles). An option is provided to disable this behaviour in static.config.js

## Types of changes

- [ ] Refactoring/add tests (refactoring or adding test which isn't a fix or add a feature)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly.
- [ ] My changes have tests around them

### CODE REVIEW

File: src/static/getConfig.js [hunk 1]
```diff
@@ -212,6 +212,11 @@ export const buildConfigation = (config = {}) => {
   process.env.REACT_STATIC_PREFETCH_RATE = finalConfig.prefetchRate
   process.env.REACT_STATIC_DISABLE_ROUTE_INFO_WARNING = finalConfig.disableRouteInfoWarning
   process.env.REACT_STATIC_DISABLE_ROUTE_PREFIXING = finalConfig.disableRoutePrefixing
+  process.env.REACT_STATIC_BUILD_HASH =
+    finalConfig.buildHash ||
+    Math.random()
+      .toString(36)
+      .substr(2, 9)
```
Comment (tannerlinsley) [inline src/static/getConfig.js]: Would you rather use something like the "shortid" package for this?

Comment (Rusfighter) [reply src/static/getConfig.js]: Done!

**Comment by Rusfighter** [comment]
Is it possible to merge this into 5.9.x? Somehow i cant run the tests on my windows machine

**Comment by lottamus** [comment]
Travis-CI tests are working again. Can you merge master in to this PR?

**Comment by Rusfighter** [comment]
@lottamus Done!

**Comment by tyanas** [comment]
@Rusfighter thanks for this PR! 🎉 

@lottamus JFYI my project is built nicely with react-static@master (getting all these `routeInfo.<hash>.json` files)
Can't wait for a v5.x.x release! 🙂 

Very much appreciate your work here! And ❤️ react-static

**Comment by Rusfighter** [comment]
Glad to hear, don't forget to turn on .json caching ;)

**Comment by xralphack** [comment]
@ilijaNL How to turn on .json caching?

**Comment by ilijaNL** [comment]
> @ilijaNL How to turn on .json caching?

It depends on your webhosting/CDN config. As i remember I used firebase hosting in the past. In the config you could specify what kind of file extensions you wanted to cache and for how long.

**Pull request merged**